Sources/XcodeProj/Scheme/XCScheme+TestableReference.swift

```diff
@@ -8,6 +8,7 @@ extension XCScheme {
         public var skipped: Bool
         public var parallelizable: Bool
         public var randomExecutionOrdering: Bool
+        public var useTestSelectionWhitelist: Bool
```

Comment:
Hi @basvankuijck. Thanks for opening this PR. Would you mind making this attribute optional? Since Xcode is able to open projects where this attribute is not defined that means that the attribute is optional and we should reflect it as such.

Reply:
Hi @pepibumur, I don’t understand your reasoning? If this property is not in a project it will be set to false when reading. When writing it will only be set if true. This is the same for the other booleans in this class and many others

Reply:
If a read project has that property as nil, we should keep it as nil to write it back as nil. If we turn nil into a default boolean, we’ll end up creating an attribute when we shouldn’t. That causes diff issues that users will end up reporting an issues.

It’s not an issue when generating projects, but it is when reading and writing existing projects.

Reply:
In other words, nil != false

Reply:
If a project is read where this is nil, then the property can be set to false. When the project is written again, the value won't be written if the value is false, meaning it will end up the same. This causes 0 diffs.

The only theoretical diff is if a project had this explicitly set to false, then reading and writing it would remove the property. This wouldn't happen though as Xcode removes the property when you set it to false in the UI, which is what we are proposing XcodeProj do as well

Reply:
This is also the same logic that the other bools in this class have been using for years

Reply:
This wouldn't happen though as Xcode removes the property when you set it to false in the UI, which is what we are proposing XcodeProj do as well

Xcode writes are optimized for minimizing the diffs after writing the project. If we read, we write nil and as a user of the tool there are no unexpected diffs. We might have missed this in some other properties.